server/game/cards/characters/01/alannysgreyjoy.js

```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../../drawcard.js');
+
+class AlannysGreyjoy extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: player.firstPlayer,
```

Comment:
`player` doesn't exist in this context so it will crash. Also, condition must always be a function (and needs to be in this case anyway since first player can change each round). So you want this instead:
```javascript
condition: () => this.controller.firstPlayer,
```

Otherwise, looks good to me.